shell/platform/darwin/macos/framework/Source/FlutterCompositor.h

```diff
@@ -40,12 +40,29 @@ class FlutterCompositor {
   // The view_provider is used to query FlutterViews from view IDs,
   // which are used for presenting and creating backing stores.
   // It must not be null, and is typically FlutterViewEngineProvider.
+  //
+  // The `main_queue` must be the event queue given to `FlutterThreadSynchronizer`,
+  // which is where the views' `FlutterSurfaceManager` invokes the callback in
+  // `presentSurfaces:atTime:notify:`. Typically `dispatch_get_main_queue()`.
   FlutterCompositor(id<FlutterViewProvider> view_provider,
                     FlutterTimeConverter* time_converter,
-                    FlutterPlatformViewController* platform_views_controller);
+                    FlutterPlatformViewController* platform_views_controller,
+                    dispatch_queue_t main_queue);
```

Comment:
Is there ever a case where this isn't the main queue returned by `dispatch_get_main_queue()`? If not, I think we should just test `dispatch_get_current_queue() == dispatch_get_main_queue()` where necessary.

Reply:
I'm mostly thinking about testability, although there really aren't many tests for `FlutterCompositor`. However, the `ThreadSynchronizer` accepts the queue as a replaceable parameter (which was added when I wrote tests for it). I think it's a clearer design if the same queue is passed in for both classes explicitly in production code.

Reply:
Ok, `ThreadSynchronizer` has two initializers, one of which uses the main queue implicitly. Still, its internal implementation allows a replaceable queue. I'm ok with any suggestion.